lib/AST/ASTPrinter.cpp

```diff
@@ -2461,7 +2461,7 @@ void PrintAST::printParameterList(ParameterList *PL, Type paramListTy,
                                   bool isCurried,
                                   std::function<bool()> isAPINameByDefault) {
   SmallVector<ParameterTypeFlags, 4> paramFlags;
-  if (paramListTy) {
+  if (paramListTy && !paramListTy.getPointer()->hasError()) {
```

Comment:
Type has an overloaded `operator->`, so you can just use `paramListTy->hasError()`.